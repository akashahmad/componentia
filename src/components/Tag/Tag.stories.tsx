import * as React from "react";

import type { Meta, StoryObj } from "@storybook/react";

import { Tag } from "./Tag";

const meta: Meta<typeof Tag> = {
    component: Tag,
    tags: ["autodocs"],
};

export default meta;
type Story = StoryObj<typeof Tag>;

export const Default: Story = {
    args: {
        disabled: false,
    },
    argTypes: {},
    render: props => <Tag {...props}>Tag</Tag>,
};
