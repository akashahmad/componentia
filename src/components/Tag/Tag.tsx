import * as React from "react";

import cx from "classnames";

import { Typography } from "../Typography/Typography";

export function Tag({ children, disabled = false }: TagProps) {
    return (
        <button
            disabled={disabled}
            className={cx(
                "py-[5px] px-[14px] rounded-md text-black inline-block cursor-pointer box-border",
                "focus:border-stroke focus:border-2 focus:bg-grey-9 focus:text-black",
                {
                    ["bg-grey-3 text-primary-text cursor-not-allowed"]: disabled,
                    ["bg-grey-9 hover:bg-primary hover:text-white"]: !disabled,
                },
            )}
        >
            <Typography variant="body2">{children}</Typography>
        </button>
    );
}

interface TagProps {
    children?: React.ReactNode;
    disabled?: boolean;
}
