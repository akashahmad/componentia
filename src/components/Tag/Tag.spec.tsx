import * as React from "react";

import { render } from "@testing-library/react";
import { Tag } from "./Tag";

test("Tag component is rendering properly", async () => {
    const { container } = render(<Tag />);

    expect(container).toMatchSnapshot();
});
