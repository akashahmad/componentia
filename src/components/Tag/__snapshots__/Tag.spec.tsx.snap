// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component is rendering properly 1`] = `
<div>
  <button
    class="py-[5px] px-[14px] rounded-md text-black inline-block cursor-pointer box-border focus:border-stroke focus:border-2 focus:bg-grey-9 focus:text-black bg-grey-9 hover:bg-primary hover:text-white"
  >
    <p
      class="text-base leading-6 font-normal"
    />
  </button>
</div>
`;
